uamqp/cbs.py

```diff
@@ -51,18 +49,19 @@ def __init__(
             status_code_field=b'status-code',
             status_description_field=b'status-description'
         )  # type: ManagementLink
-        self._auth_audience = auth_audience
-        self._encoding = encoding
-        self._auth_timeout = auth_timeout
 
-        if not get_token or not callable(get_token):
+        if not auth.get_token or not callable(auth.get_token):
             raise ValueError("get_token must be a callable object.")
 
-        self._get_token = get_token
-        self._token_type = token_type
+        self._auth = auth
+        self._audience = self._auth.audience
+        self._get_token = self._auth.get_token
+        self._token_type = self._auth.token_type
```

Comment:
Do we need to store these in local variables when we've already stored `self._auth`?

Reply:
nope, thanks for pointing out, just removed those ivars